webrtc/RTCTransport-helper.js

```diff
@@ -0,0 +1,184 @@
+    'Expect pc.sctp to be instantiated from RTCSctpTransport');
+
+  const dtlsTransport = sctpTransport.transport;
+  assert_true(dtlsTransport instanceof RTCDtlsTransport,
+    'Expect sctp.transport to be an RTCDtlsTransport');
+
+  return dtlsTransport;
+}
+
+function getIceTransportFromSctpTransport(sctpTransport) {
+  const dtlsTransport = getDtlsTransportFromSctpTransport(sctpTransport);
+
+  const iceTransport = dtlsTransport.transport;
+  assert_true(iceTransport instanceof RTCIceTransport,
+    'Expect dtlsTransport.transport to be an RTCIceTransport');
+
+  return iceTransport;
+}
+
+// Get one or more RTCDtlsTransports from RTCRtpSender/Receiver
+// in pc. There may be more than one e.g. there are more than one
+// senders/receivers or when there are different underlying
+// transports for RTP and RTCP.
+function getDtlsTransportsFromSenderReceiver(pc) {
+  const senders = pc.getSenders();
+  const receivers = pc.getReceivers();
+
+  const dtlsTransportsSet = new Set();
+
+  function adddDtlsTransport(transport) {
+    // Add a dtls transport to the result set, if it is
+    // not null/undefined. Although the spec mandates both
+    // transport and rtcpTransport fields must be set together,
+    // we validate that requirement in separate test cases.
+    if (transport) {
+      assert_true(transport instanceof RTCDtlsTransport,
+        'Expect transport to be an RTCDtlsTransport');
+
+      dtlsTransportsSet.add(transport);
+    }
+  }
+
+  for (const sender of senders) {
+    adddDtlsTransport(sender.transport);
+    adddDtlsTransport(sender.rtcpTransport);
+  }
+
+  for (const receiver of receivers) {
+    adddDtlsTransport(receiver.transport);
+    adddDtlsTransport(receiver.rtcpTransport);
+  }
+
+  const dtlsTransports = [...dtlsTransportsSet];
+
+  if (dtlsTransports.length === 0) {
+    assert_unreached('Expect to get at least one unique RTCDtlsTransport from sender receivers');
+  }
+
+  return dtlsTransports;
+}
+
+function getIceTransportFromDtlsTransport(dtlsTransport) {
+  const iceTransport = dtlsTransport.transport;
+
+  assert_true(iceTransport instanceof RTCIceTransport,
+    'Expect dtlsTransport.transport to be an RTCIceTransport');
+
+  return iceTransport;
+}
+
+function getIceTransportsFromDtlsTransports(dtlsTransports) {
+  const iceTransports = new Set();
+
+  for (const dtlsTransport of dtlsTransports) {
+    iceTransports.add(getIceTransportFromDtlsTransport(dtlsTransport));
+  }
+
+  return [...iceTransports];
+}
+
+function getIceTransportsFromSenderReceiver(pc) {
+  const dtlsTransports = getDtlsTransportsFromSenderReceiver(pc);
+
+  const iceTransports = new Set(
+    dtlsTransports.map(getIceTransportFromDtlsTransport));
+
+  return [...iceTransports];
+}
+
+// Create DTLS transports by creating data channels
+// and connecting two peer connections. Returns an
+// array of two RTCDtlsTransports, which obtained
+// from RTCSctpTransport in pc.sctp.
+function createDtlsTransportsFromSctp(pc1, pc2) {
```

Comment:
Suggest renaming this function "connectPeerConnectionsAndReturnDtlsTransports".

I'd really prefer to not have this function, but instead have "connectPeerConnnections" and the (existing) getDtlsTransportFromSctpTransport called from the test functions - merging these two disparate things makes the test harder to read.